### ISSUE TITLE
SVNPoller stops abruptly

### ISSUE THREAD

**Issue by aj062** [buildbot/buildbot#3646]
I noticed that my buildbot instance (version 0.9.11) wasn't picking up changes from SVN repository. On looking at twistd.log files, I noticed that there was no SVNPoller related logs recently. It was working fine till yesterday and it stopped abruptly without any error messages in the logs.

These are the logs I see related to SVNPoller:
```
[~] grep SVNPoller twistd.log.1
2017-09-21 23:53:16-0700 [-] SVNPoller: _process_changes 2379 .. 2379
2017-09-21 23:53:16-0700 [-] SVNPoller: finished polling None
2017-09-21 23:54:12-0700 [-] SVNPoller: polling
2017-09-21 23:54:12-0700 [-] SVNPoller: repourl=https://myproject, root=https://myproject, so prefix=
2017-09-21 23:54:15-0700 [-] SVNPoller: no changes
2017-09-21 23:54:15-0700 [-] SVNPoller: _process_changes 2379 .. 2379
2017-09-21 23:54:15-0700 [-] SVNPoller: finished polling None
2017-09-21 23:55:12-0700 [-] SVNPoller: polling

[~] grep SVNPoller twistd.log
[~] 

```

It seems from logs like SVNPoller started polling, but never finished. Can it be stuck somewhere?
Can the network request (svn info --xml --non-interactive  https://svn-repository) in https://git.io/vdJTw  hang forever in case of a network issue?

**Comment by aakashja** [comment]
Some people tell me that they have seen such issue with buildbot 0.8 as well.

**Comment by tardyp** [comment]
I am not sure if we have a lot of svnpoller experts remaining here.. most of us has moved to git, and hooks. Any reason why not to switch to a svn hook?

I know that llvm guys are using that right now @chapuni may help

**Comment by seankelly** [comment]
Anything else in the log after that?

**Comment by aj062** [comment]
Nope, no other logs. It was in fact "svn info" command which was stuck forever. Killing the svn process solves the problem temporarily.

Configuring a timeout in svn seems to fix the problem.

I don't know if it is easy for buildbot to implement a timeout on 'svn info' command at https://git.io/vdJTw, Buildbot seems to use Twisted (twisted.internet.utils.getProcessOutput) to execute that command.

**Comment by p12tic** [comment]
Fixed in 7336e78a659007b68d6c9c703566d056863ed7f4, which landed to v3.2.0.
